### DISCUSSION TITLE
Vue/Inertija/Vite don't work with @routes?

### DISCUSSION THREAD

**Discussion by niels-numbers** [tighten/ziggy#688] [Q&A] [answered]
I want to use Ziggy with Laravel / Inertia / Vue / VIte

This is what I did:

1. Installed Ziggy: composer require tightenco/ziggy
2. Setup @routes in my app.blade.php:

```
<!DOCTYPE html>
<html class="h-full bg-white">
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0" />
    @routes
    @vite(['resources/assets/css/tailwindcore.css', 'resources/assets/js/tailwind-colorful/dashboard/app.js'])
    @inertiaHead
</head>
<body class="h-full">
@inertia
</body>
</html>
```

Inside my Vue Component I use the route helper like this:

```
 <td class="relative whitespace-nowrap py-4 pl-3 pr-4 text-right text-sm font-medium sm:pr-0">
                <a :href="route('admin.congresses.edit', {congress: congress.id})" class="text-cyan-600 hover:text-cyan-900"
                >Edit<span class="sr-only">, {{  congress.title }}</span></a
                >
 </td>
 ```

This fails with 

Uncaught (in promise) TypeError: _ctx.route is not a function

There is some explanation how to setup frameworks without @routes, but it also states at https://github.com/tighten/ziggy#javascript-frameworks that most don't need this. Is this not true for vite/vue/inertija? Do I have to follow the instructions there?

**Comment by niels-numbers** [comment]
I managed to get it working using the https://github.com/tighten/ziggy#javascript-frameworks, but maybe if vite, vue, inertija don't work with @routes, a note would be great in the docs. Can someone confirm this is not supposed to work?

**Comment by bakerkretzmar** [comment, marked as answer]
Inertia and Vue work fine with `@routes`. Making _any_ global JavaScript function/variable available inside the `<template>` of Vue single-file components, whether that's `route()` or `console.log()` or anything else, requires some additional setup. You can either do that setup yourself or you can use Ziggy's Vue plugin, which does it for you.

Inside the `<script>` of your components, and anywhere else in your JS code, `route()` should be available with the setup you showed above.

I agree that note in the docs is a bit misleading, I'll tweak that wording.
